### PR TITLE
ef_lamp_light OK

### PR DESCRIPTION
**Pull request by MegaIDK**

### CODE REVIEW

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 1]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
     /* */ Ef_Lamp_Light_actor_ct,
     /* */ Ef_Lamp_Light_actor_dt,
     /* */ Ef_Lamp_Light_actor_move,
-    /* */ (ActorFunc)none_proc1,
+    /* */ (void*)none_proc1,
     /* */ NULL,
 };
-#endif
+static s16 D_80967884_jp[4] = { 2, 2, 1, 0 };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:31]:
```suggestion
static s16 add_data_off[4] = { 2, 2, 1, 0 };
```

Is the static neccessary though

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 2]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+static s16 D_8096788C_jp[4] = { 0x10, 0x10, 8, 0 };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:32]:
```suggestion
static s16 add_data_on[4] = { 0x10, 0x10, 8, 0 };
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 3]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+static s16* D_80967894_jp[2] = { D_80967884_jp, D_8096788C_jp };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:33]:
```suggestion
static s16* add_data[2] = { D_80967884_jp, D_8096788C_jp };
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 4]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+static s16 D_8096789C_jp[4] = { 0, 0, 0, 0 };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:34]:
```suggestion
static s16 cmp_data_off[4] = { 0, 0, 0, 0 };
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 5]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+static s16 D_809678A4_jp[4] = { 0xC8, 0xC8, 0x96, 0 };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:35]:
```suggestion
static s16 cmp_data_on[4] = { 0xC8, 0xC8, 0x96, 0 };
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 6]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+static s16* D_809678AC_jp[2] = { D_8096789C_jp, D_809678A4_jp };
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:36]:
```suggestion
static s16* cmp_data[2] = { D_8096789C_jp, D_809678A4_jp };
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 7]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+}
+
+void Ef_Lamp_Light_actor_dt(Actor* thisx, Game_Play* game_play) {
+    Lamp_Light* this = (Lamp_Light*)thisx;
+
+    game_play->kankyo.unk_1C = NULL;
+    Global_light_list_delete(&game_play->glight, this->unk_188);
+}
+
+s32 func_80967644_jp(void) {
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:54]:
```suggestion
s32 eLL_get_light_sw_other_room(void) {
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 8]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+
+    temp_v0 = common_data.time.nowSec;
+    var_v1 = 0;
+    if ((temp_v0 < 0x5460) || (temp_v0 >= 0xFD20)) {
+        var_v1 = 1;
+    }
+    return var_v1;
+}
+
+s32 func_80967678_jp(Lamp_Light* this UNUSED) {
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:66]:
```suggestion
s32 eLL_get_light_sw_player_room(Lamp_Light* this UNUSED) {
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 9]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+    s32 var_v1;
+    s32 temp_v0;
+
+    temp_v0 = common_data.time.nowSec;
+    var_v1 = ((temp_v0 >= 0xFD20) && (temp_v0 < 0x14370)) ? 1 : 0;
+    return var_v1;
+}
+
+s32 func_809676BC_jp(void) {
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:75]:
```suggestion
s32 eLL_get_light_sw_start_demo(void) {
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 10]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+    s32 var_v1;
+
+    var_v1 = 1;
+    if ((u8)common_data.unk_104AE[0] == 1) {
+        var_v1 = 0;
+    }
+    return var_v1;
+}
+
+void func_809676E4_jp(Lamp_Light* this) {
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:85]:
```suggestion
void eLL_ctrl_light_sw(Lamp_Light* this) {
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 11]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+    s32 var_v1 = 0;
+
+    switch (common_data.unk_10001) {
+        case 2:
+        case 3:
+            switch (mFI_GetFieldId()) {
+                case 0x6000:
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:92]: These are from the player enum, but I can change this later when I merge cockroach

File: include/m_kankyo.h [hunk 12]
```diff
@@ -15,7 +17,9 @@ typedef struct Kankyo {
     /* 0x03 */ s8 unk_03;
     /* 0x04 */ s8 unk_04;
     /* 0x05 */ UNK_TYPE1 unk_05[0x3];
-    /* 0x08 */ UNK_TYPE1 unk_08[0xA6];
+    /* 0x08 */ UNK_TYPE1 unk_08[0x14];
+    /* 0x1C */ Lights* unk_1C;
```
Comment (AngheloAlf) [inline include/m_kankyo.h:21]:
```suggestion
    /* 0x1C */ struct Lights* unk_1C;
```

File: include/m_kankyo.h [hunk 13]
```diff
@@ -3,9 +3,11 @@
 
 #include "ultra64.h"
 #include "unk.h"
+#include "m_lights.h"
```
Comment (AngheloAlf) [inline include/m_kankyo.h:7]:
```suggestion
#include "unk.h"

```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 14]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+static s16 D_8096789C_jp[4] = { 0, 0, 0, 0 };
+static s16 D_809678A4_jp[4] = { 0xC8, 0xC8, 0x96, 0 };
+static s16* D_809678AC_jp[2] = { D_8096789C_jp, D_809678A4_jp };
+
+void Ef_Lamp_Light_actor_ct(Actor* thisx, Game_Play* game_play) {
+    s32 pad UNUSED;
+    Lamp_Light* this = (Lamp_Light*)thisx;
+
+    game_play->kankyo.unk_1C = &this->unk_178;
+    Light_diffuse_ct(&this->unk_178, 0, 0x50, 0, (u8)0, (u8)0, (u8)0);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:43]: Are the casts required?
```suggestion
    Light_diffuse_ct(&this->unk_178, 0, 0x50, 0, 0, 0, 0);
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 15]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+    temp_v0 = common_data.time.nowSec;
+    var_v1 = ((temp_v0 >= 0xFD20) && (temp_v0 < 0x14370)) ? 1 : 0;
+    return var_v1;
+}
+
+s32 func_809676BC_jp(void) {
+    s32 var_v1;
+
+    var_v1 = 1;
+    if ((u8)common_data.unk_104AE[0] == 1) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:79]: This seems to be a case where you need to change the member in `CommonData`
`unk_104AE` currently is an array, but seems like it should be changed to a single member of type `u8`. Don't forget to add extra members to account for the size of the struct

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c [hunk 16]
```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
...
+    this->unk_174 = var_v1;
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Lamp_Light/ef_lamp_light/func_809676BC_jp.s")
+void Ef_Lamp_Light_actor_move(Actor* thisx, Game_Play* game_play UNUSED) {
+    Lamp_Light* this = (Lamp_Light*)thisx;
+    s16* var_s1;
+    s16* var_s2;
+    s32 temp_v1;
+    s32 var_s3;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:123]: The for loop could be a bit more readable if you name this variable `i`
```suggestion
    s32 i;
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.h [hunk 17]
```diff
@@ -12,7 +13,10 @@ typedef void (*Lamp_LightActionFunc)(struct Lamp_Light*, struct Game_Play*);
 
 typedef struct Lamp_Light {
     /* 0x000 */ Actor actor;
-    /* 0x174 */ UNK_TYPE1 unk_174[0x18];
+    /* 0x174 */ s32 unk_174;
+    /* 0x178 */ Lights unk_178;
+    /* 0x174 */ char unk_186[0x2];
+    /* 0x188 */ LightNode* unk_188;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.h:19]: `unk_186` seems to be just struct padding, so you could just remove it
```suggestion
    /* 0x178 */ Lights unk_178;
    /* 0x188 */ LightNode* unk_188;
```

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c (hunk 1 above)
Comment (emilybrooks) [reply src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:31]: they appear to be in function static in Ef_Lamp_Light_actor_move, so I would do that

File: src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c (hunk 7 above)
Comment (MegaIDK) [reply src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:54]: should i change the other ones from `Ef_Lamp_Light` to `eLL` as well? like `Ef_Lamp_Light_actor_ct` to `eLL_actor_ct`

Comment (Prakxo) [reply src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c:54]: They're genuinely the original names lol, no don't change it

**Pull request merged**